stan/math/prim/arr/functor/map_rect.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#ifdef STAN_HAS_MPI
+#include <stan/math/prim/arr/functor/map_rect_mpi.hpp>
+#endif
+
+namespace stan {
+  namespace math {
+
+    template <typename F>
+    std::vector<double>
```

Comment:
Can you not unify this with the one in `rev` by using a template instead of `double` here (and instead of `var` in the `rev` version)? They seem to be identical otherwise.

Reply:
hmm... I think you are right in that we can template this and have a single version. Not sure why I did it this way (I added the map_rect quite late).

However, I hope this does not interact badly with the function overloading which we do for the mpi version.